### PR TITLE
chore(test): Add unit test to chaos runner

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit <udit@chaosnative.com>
**What this PR does / why we need it**:

- This is for increasing the unit test coverage for the chaos runner repository to more than `50%` from `13%` . 
- The unit test for the following files has been added:
- `pkg/utils/common_test.go`
- `pkg/utils/configMapUtils_test.go`
- `pkg/utils/engineHelper_test.go`
- `pkg/utils/event_test.go`
- `pkg/utils/experimentHelper_test.go`
- `pkg/utils/secretsUtils_test.go`
- `pkg/utils/watchChaosContainer_test.go`
- `pkg/utils/watchJob_test.go`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by uditgaurav** [comment]
### Coverage:

- `go test ./... -coverprofile=coverage.txt -covermode=atomic -v -count=1`

```
...
time="2021-03-06T18:32:44+05:30" level=info msg="deleting the job as jobCleanPolicy is set to delete"
=== RUN   TestDeleteJobAccordingToJobCleanUpPolicy/Test_Positive-2
=== RUN   TestDeleteJobAccordingToJobCleanUpPolicy/Test_Negative-1
time="2021-03-06T18:32:44+05:30" level=info msg="deleting the job as jobCleanPolicy is set to "
--- PASS: TestDeleteJobAccordingToJobCleanUpPolicy (0.00s)
    --- PASS: TestDeleteJobAccordingToJobCleanUpPolicy/Test_Positive-1 (0.00s)
    --- PASS: TestDeleteJobAccordingToJobCleanUpPolicy/Test_Positive-2 (0.00s)
    --- PASS: TestDeleteJobAccordingToJobCleanUpPolicy/Test_Negative-1 (0.00s)
PASS
coverage: 56.7% of statements
ok  	github.com/litmuschaos/chaos-runner/pkg/utils	34.055s	coverage: 56.7% of statements
```

File: pkg/utils/configMapUtils_test.go [hunk 1]
```diff
@@ -148,3 +149,455 @@ func TestPatchConfigMaps(t *testing.T) {
...
+			client := CreateFakeClient(t)
+
+			if !mock.isErr {
+				_, err := client.KubeClient.CoreV1().ConfigMaps(experiment.Namespace).Create(&mock.configmap)
+				if err != nil {
+					t.Fatalf("configmap not created for %v test, err: %v", name, err)
+				}
+			}
+			if mock.isErr {
+				_, err := client.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(mock.chaosexperiment.Namespace).Create(mock.chaosexperiment)
```
Comment (ispeakc0de) [inline pkg/utils/configMapUtils_test.go:302]: Do we need experiment CR creation here?

File: pkg/utils/configMapUtils_test.go [hunk 2]
```diff
@@ -148,3 +149,455 @@ func TestPatchConfigMaps(t *testing.T) {
...
+		EngineNamespace: "Fake NameSpace",
+	}
+
+	tests := map[string]struct {
+		chaosexperiment *litmuschaosv1alpha1.ChaosExperiment
+		chaosengine     *litmuschaosv1alpha1.ChaosEngine
+		isErr           bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
```
Comment (ispeakc0de) [inline pkg/utils/configMapUtils_test.go:453]: Do we need engine CR here?

File: pkg/utils/event_test.go [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+		Type:    "fake-type",
+		Name:    "fake-name",
+	}
+
+	tests := map[string]struct {
+		chaosengine *litmuschaosv1alpha1.ChaosEngine
+		isErr       bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:36]: Do we need chaosengine here?

File: pkg/utils/event_test.go [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+		Name:    "fake-name",
+	}
+
+	tests := map[string]struct {
+		chaosengine *litmuschaosv1alpha1.ChaosEngine
+		events      v1.Event
+		isErr       bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:91]: do we need engine here?

File: pkg/utils/event_test.go [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+}
+
+func TestExperimentSkipped(t *testing.T) {
+	engineDetails := EngineDetails{
+		Name:            "Fake Engine",
+		EngineNamespace: "Fake NameSpace",
+		UID:             "",
+	}
+
+	eventAtr := EventAttributes{
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:165]: Do we need eventAtr here?

File: pkg/utils/event_test.go [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+		StatusCheckTimeout: 2,
+	}
+
+	tests := map[string]struct {
+		chaosengine *litmuschaosv1alpha1.ChaosEngine
+		events      v1.Event
+		isErr       bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:184]: Do we need chaosengine here?

File: pkg/utils/event_test.go [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+}
+
+func TestExperimentDependencyCheck(t *testing.T) {
+	engineDetails := EngineDetails{
+		Name:            "Fake Engine",
+		EngineNamespace: "Fake NameSpace",
+		UID:             "",
+	}
+
+	eventAtr := EventAttributes{
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:259]: same as above, do the same for others

File: pkg/utils/experimentHelper_test.go [hunk 8]
```diff
@@ -0,0 +1,861 @@
...
+		Name:               "Fake-Exp-Name",
+		Namespace:          "Fake NameSpace",
+		JobName:            "fake-job-name",
+		StatusCheckTimeout: 10,
+		envMap:             map[string]corev1.EnvVar{},
+	}
+
+	tests := map[string]struct {
+		chaosexperiment *litmuschaosv1alpha1.ChaosExperiment
+		chaosengine     *litmuschaosv1alpha1.ChaosEngine
```
Comment (ispeakc0de) [inline pkg/utils/experimentHelper_test.go:102]: Do we need engine here?

File: pkg/utils/experimentHelper_test.go [hunk 9]
```diff
@@ -0,0 +1,861 @@
...
+
+	engineDetails := EngineDetails{
+		Name:            "Fake Engine",
+		EngineNamespace: "Fake NameSpace",
+		UID:             "fake-chaosuid",
+	}
+
+	tests := map[string]struct {
+		chaosexperiment *litmuschaosv1alpha1.ChaosExperiment
+		chaosengine     *litmuschaosv1alpha1.ChaosEngine
```
Comment (ispeakc0de) [inline pkg/utils/experimentHelper_test.go:197]: same as above, remove from other places as well

File: pkg/utils/configMapUtils_test.go [hunk 10]
```diff
@@ -148,3 +149,438 @@ func TestPatchConfigMaps(t *testing.T) {
...
+	fakeConfigMap := "fake configmap"
+	experiment := ExperimentDetails{
+		Name:               "Fake-Exp-Name",
+		Namespace:          "Fake NameSpace",
+		JobName:            "fake-job-name",
+		StatusCheckTimeout: 10,
+	}
+
+	tests := map[string]struct {
+		chaosexperiment *litmuschaosv1alpha1.ChaosExperiment
```
Comment (ispeakc0de) [inline pkg/utils/configMapUtils_test.go:259]: Do we need chaosexperiment here?

Comment (uditgaurav) [reply pkg/utils/configMapUtils_test.go:259]: We have an empty spec for chaosengine

File: pkg/utils/event_test.go (hunk 5 above)
Comment (uditgaurav) [reply pkg/utils/event_test.go:119]: We are using it in `events` Spec

File: pkg/utils/event_test.go (hunk 7 above)
Comment (uditgaurav) [reply pkg/utils/event_test.go:198]: We are using it in `events` Spec

File: pkg/utils/configMapUtils_test.go [hunk 11]
```diff
@@ -148,3 +149,384 @@ func TestPatchConfigMaps(t *testing.T) {
...
+		chaosengine     *litmuschaosv1alpha1.ChaosEngine
+		isErr           bool
+	}{
+		"Test Positive-1": {
+			chaosengine: &litmuschaosv1alpha1.ChaosEngine{},
```
Comment (ispeakc0de) [inline pkg/utils/configMapUtils_test.go:431]: Do we need engine here?

File: pkg/utils/event_test.go [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+	}
+	experiment := ExperimentDetails{
+		Name:               "Fake-Exp-Name",
+		Namespace:          "Fake NameSpace",
+		JobName:            "fake-jobs-name-12345",
+		StatusCheckTimeout: 2,
+	}
+
+	tests := map[string]struct {
+		chaosengine *litmuschaosv1alpha1.ChaosEngine
```
Comment (ispeakc0de) [inline pkg/utils/event_test.go:131]: we can remove this

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**